### PR TITLE
For #7887 fix flaky verifyDownloadedFileOnStorageTest UI test

### LINKED ISSUE #7887: Intermittent UI test failure - < DownloadFileTest.verifyDownloadedFileOnStorageTest>
### Stacktrace:
java.lang.AssertionError: textfile(2).txt finished
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.mozilla.focus.activity.robots.DownloadRobot.verifyDownloadConfirmationMessage(DownloadRobot.kt:51)
	at org.mozilla.focus.activity.DownloadFileTest$verifyDownloadedFileOnStorageTest$3.invoke(DownloadFileTest.kt:205)
	at org.mozilla.focus.activity.DownloadFileTest$verifyDownloadedFileOnStorageTest$3.invoke(DownloadFileTest.kt:202)
	at org.mozilla.focus.activity.robots.DownloadRobotKt.downloadRobot(DownloadRobot.kt:67)
	at org.mozilla.focus.activity.DownloadFileTest.verifyDownloadedFileOnStorageTest(DownloadFileTest.kt:202)

### PR DESCRIPTION
**Pull request by AndiAJ**
For #7887 fix flaky `verifyDownloadedFileOnStorageTest` UI test ✅ successfully passed 100x. on Firebase

Summary:
For verifying that the downloaded file is actually saved on the device, we used similar code with what we [have in AC to handle downloads.](https://github.com/mozilla-mobile/android-components/blob/6dc07542819cfb29b4bf2a77f2eb8df37f1d54ea/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L963-L1007) 

Rarely, the test was flaky, failing to verify that the file is actually saved on the device storage (not sure why the returned download uri was null)

As an alternative, @Mugurell suggested to try using the implementation from [this tutorial](https://programmerworld.co/android/how-to-check-if-a-file-exists-in-download-folder-from-your-android-app-api-level-30-android-11-and-above/)

I made the changes based on our interest, and it seems to be working properly.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### GitHub Automation
Fixes #7887

### CODE REVIEW

**Comment by AndiAJ** [comment]
Adding @Mugurell as a reviewer, to share his thoughts as well. 🤞🏽

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt [hunk 1]
```diff
@@ -176,66 +174,12 @@ object TestHelper {
...
-            if (it.count > 0) {
-                val idColumnIndex = it.getColumnIndex(MediaStore.Downloads._ID)
-                it.moveToFirst()
-                downloadUri = ContentUris.withAppendedId(collection, it.getLong(idColumnIndex))
-            }
-        }
-
-        return downloadUri
+        val storageManager: StorageManager? =
+            getInstrumentation().targetContext.getSystemService() as StorageManager?
```
Comment (Mugurell) [inline app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt:178]: nit: The safe call operator - `?` is not needed here.

```suggestion
val storageManager = getInstrumentation().targetContext.getSystemService(Context.STORAGE_SERVICE) as StorageManager
```

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt [hunk 2]
```diff
@@ -176,66 +174,12 @@ object TestHelper {
...
+        val storageVolumes = storageManager!!.storageVolumes
+        val storageVolume: StorageVolume = storageVolumes.get(0)
+        val file = File(storageVolume.directory!!.path + "/Download/" + fileName)
```
Comment (Mugurell) [inline app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt:181]: nit: In kotlin a template is preferable over concatenation
```suggestion
        val file = File("${storageVolume.directory!!.path}/Download/$filename")
```

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt [hunk 3]
```diff
@@ -176,66 +174,12 @@ object TestHelper {
...
-                it.moveToFirst()
-                downloadUri = ContentUris.withAppendedId(collection, it.getLong(idColumnIndex))
-            }
-        }
-
-        return downloadUri
+        val storageManager: StorageManager? =
+            getInstrumentation().targetContext.getSystemService() as StorageManager?
+        val storageVolumes = storageManager!!.storageVolumes
+        val storageVolume: StorageVolume = storageVolumes.get(0)
```
Comment (Mugurell) [inline app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt:180]: nit: An indexing operator might be more idiomatic.

```suggestion
        val storageVolume: StorageVolume = storageVolumes[0]
```

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt (hunk 2 above)
Comment (Mugurell) [inline app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt:181]: The `storageVolume.directory` required API 30 which I understand is okay to use in our automatic tests.
Adding the `@RequiresApi(VERSION_CODES.R)` annotation to the `verifyDownloadedFileOnStorage` method might help prevent unintended usages.

**Review by Mugurell — APPROVED** [review]
Rubberstamping if this helps with more stable tests.

**Comment by AndiAJ** [comment]
Thanks @Mugurell , made the changes as suggested.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**